cmd/internal/cli/key_import.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2017-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/sylabs/singularity/docs"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+	"github.com/sylabs/singularity/pkg/sypgp"
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/errors"
+)
+
+func init() {
+	KeyImportCmd.Flags().SetInterspersed(false)
+}
+
+// KeyImportCmd is `singularity key (or keys) import` and imports a local key into the singularity key store.
+var KeyImportCmd = &cobra.Command{
+	Args:                  cobra.ExactArgs(1),
+	DisableFlagsInUseLine: true,
+	PreRun:                sylabsToken,
+	Run:                   importRun,
+	Use:                   docs.KeyImportUse,
+	Short:                 docs.KeyImportShort,
+	Long:                  docs.KeyImportLong,
+	Example:               docs.KeyImportExample,
+}
+
+func doKeyImportCmd(path string) error {
+
+	var fingerprint [20]byte
+
+	// PublicKeyType is the armor type for a PGP public key.
+	var PublicKeyType = "PGP PUBLIC KEY BLOCK"
+
+	// PrivateKeyType is the armor type for a PGP private key.
+	var PrivateKeyType = "PGP PRIVATE KEY BLOCK"
+
+	//open the file to check if this corresponds to a private or public key
```

Comment:
For consistently purposes, I would add a space here:

```suggestion
	// open the file to check if this corresponds to a private or public key
```